File: highest-scoring-word/highestScoringWord.js
const high = (string) => {
    const alpha = ' abcdefghijklmnopqrstuvwxyz';
    const values = string.split(' ');

    const vals = values.map((element) => {
        let val = 0;
        element.split('').forEach((letter) => {
            val += alpha.indexOf(letter);
        });
        return val;
    });

    return values[vals.indexOf(Math.max(... vals))];
};

module.exports = { high };


//https://www.codewars.com/kata/57eb8fcdf670e99d9b000272/train/javascript

File: python-venv/highest_scoring_word/kata_function/highest_scoring_word.py
def high(string):
    alpha = ' abcdefghijklmnopqrstuvwxyz'
    values = string.split(' ')

    def calc(array):
        val = 0
        for i in range(len(array)):
            val += alpha.index(array[i])
        return val

    vals = list(map(lambda word: calc(list(word)), values))

    return values[vals.index(max(vals))]